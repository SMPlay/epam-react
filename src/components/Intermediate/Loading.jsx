import React from "react";
import spinner from "../img/Spinner.gif";

const Loading = () => <img src={spinner} alt="Loading" />;

export default Loading;
