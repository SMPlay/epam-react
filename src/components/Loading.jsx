import React from "react";
import { createUseStyles } from "react-jss";
import spinner from "../img/Spinner.gif";

const useStyles = createUseStyles({
  loading: {
    height: "100vh",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
});

export const Loading = () => {
  const classes = useStyles();

  return <div className={classes.loading}><img src={spinner} alt="Loading..." /></div>;
};
