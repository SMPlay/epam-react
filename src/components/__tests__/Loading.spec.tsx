import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Loading } from "../Loading";

describe("Loading component", () => {
  it("should render", () => {
    const tree = renderer.create(
      <Router>
        <Loading />
      </Router>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
