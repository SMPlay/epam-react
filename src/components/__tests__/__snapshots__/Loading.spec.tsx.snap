// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render 1`] = `
<div
  className="loading-0-2-1"
>
  <img
    alt="Loading..."
    src="Jest mock img"
  />
</div>
`;
